### ISSUE TITLE
Example monorepo / tutorials

### ISSUE THREAD

**Issue by schickling** [Updater/semantic-release-monorepo#66]
First of all: Thanks a lot for working on this package – it looks very promising. I couldn't find any example monorepo and/or tutorials. Would you mind providing some? 🙏

**Comment by gund** [comment]
+ 1 for examples.

Also I would like to see `non-lerna` based examples (for ex `nx-workspaces`) that will suit any monorepo.

I tried to do this:
```
// Release package1
cd path/to/publishable/package1 && npx semantic-release -e semantic-release-monorepo
// Release package2
cd path/to/publishable/package2 && npx semantic-release -e semantic-release-monorepo
// ... And so on
```

But when I was doing so I got analyzer saying that it could not find any commits so it will skip release.

Example: https://travis-ci.org/orchestratora/orchestrator/jobs/444578949#L569
```
...
[Semantic release]: Found 46 commits since last release
[Semantic release]: Call plugin analyze-commits
[Semantic release]: Found 0 commits for package @orchestrator/core since last release
[Semantic release]: Analysis of 0 commits complete: no release
[Semantic release]: There are no relevant changes, so no new version is released.
```

That is why it would be nice to have more relevant examples.

Thanks =)

**Comment by pmowrer** [comment]
I totally hear you! The difficulty in creating good examples is that there are a lot of scenarios to cover!

@gund Would you mind running with the `--debug` flag? It will print a lot of additional information, including why a commit isn't considered for a release.

**Comment by marcguilera** [comment]
Is it possible to use this for non-node projects?

**Comment by snebjorn** [comment]
I too would appreciate some examples. It doesn't have to cover may cases, just a simple getting started scenario.

Cause reading the doc have me confused.

>Run `semantic-release-monorepo` for the package in the current working directory:
> > npx semantic-release -e semantic-release-monorepo

Do I run `semantic-release-monorepo` or do i run `semantic-release -e semantic-release-monorepo`. It isn't clear and I have to try both to be sure

**Comment by pmowrer** [comment]
I agree the language could be clearer @snebjorn. I'll clean it up soon. Anyway, `semantic-release -e semantic-release-monorepo` is how you run it.

**Comment by ispirals** [comment]
Hi, im facing the same issue described by @gund using yarn workspaces.

Here the ouput with --debug

```
[2:11:22 PM] [semantic-release] › ℹ  Found 18 commits since last release
  semantic-release:get-commits Parsed commits: [ {...the 18 commits found } ] +0ms
[2:11:22 PM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "[Function: semantic-release-monorepo]"
  semantic-release:monorepo Running 'analyzeCommits' version '7.0.2' +0ms
  semantic-release:monorepo Filter commits by package path: "" +0ms
[2:11:22 PM] [semantic-release] [[Function: semantic-release-monorepo]] › ℹ  Found 0 commits for package lec-licensing since last release
[2:11:22 PM] [semantic-release] [[Function: semantic-release-monorepo]] › ℹ  Analysis of 0 commits complete: no release
```

Seems a problem with the `analyzeCommits` step.

I just installed lerna in my project, and now it works as expected.

**Comment by pmowrer** [comment]
Hi @ispirals!

Based on `Filter commits by package path: ""`, it looks like you're running `semantic-release` in the root of the repo and not in the root a monorepo package. Would you mind sharing how you're running this with lerna?

**Comment by Mani2693** [comment]
Same issue with nx dev repo , I run the command from root of the project but it says 0 commit found

**Comment by pmowrer** [comment]
Hi @Mani2693! The command needs to be run in the root of each package, not in the repo (project) root.

**Comment by snebjorn** [comment]
@Mani2693 if you want to run from root see this "trick" #85

**Comment by Mani2693** [comment]
@pmowrer @snebjorn  Thank you so much for your response, I tried it from the project folder still getting same error, It was not getting commits form git , I tried with lerna exec as well still no go , Kindly help,
![image](https://user-images.githubusercontent.com/11607412/79304923-ff23e180-7f0f-11ea-87de-6a01fee9fd35.png)

**Comment by snebjorn** [comment]
I haven't tried that trick with NPM so don't know if it have the same capabilities. I also don't know lerna.

But I need to see the scripts and your configs to help further.

**Comment by Mani2693** [comment]
@snebjorn I have sample project attached here , this has firstlib which i am trying to apply release update.
[testproj.zip](https://github.com/pmowrer/semantic-release-monorepo/files/4503408/testproj.zip)

Thanks for your time !

**Comment by hd719** [comment]
Hey quick question in the monorepo directory I would run this command (if we are using pnpm) `pnpm -r --workspace-concurrency=1 exec -- npx --no-install semantic-release -e semantic-release-monorepo` 

And then in the individual package level I would run this `npm run semantic-release` ?
